src/main/java/org/testng/internal/Graph.java

```diff
@@ -121,7 +122,10 @@ public void topologicalSort() {
     // Sort the nodes alphabetically to make sure that methods of the same class
     // get run close to each other as much as possible
     //
-    Collections.sort(nodes2);
+    Comparator<Node> comparator = Systematiser.getComparator();
```

Comment:
The comparator should be an attribute of the graph and the comparator instantiation logic should be outside.
The attribute description will be: `private final Comparator<Node<T>> comparator` and its usage will be `Collections.sort(nodes2, comparator)`.